compiler/rustc_middle/src/ty/diagnostics.rs

```diff
@@ -25,6 +25,8 @@ impl<'tcx> IntoDiagnosticArg for Ty<'tcx> {
     }
 }
 
+pub struct TyParam<'a>(pub &'a str, pub Option<DefId>);
```

Comment:
`ParamTy` already exists. `TyParamInfo` isn't great either, but maybe a little less confusing?

Reply:
What do you think of `TyParamContext`?

Reply:
These values describe the type parameter itself though and not the context it is in, so I don't think `TyParamContext` is a good name. I would maybe pick `TyParamDiagnosticsInfo`. You can also just change this back to taking two parameters, it's really not that important to me.

Reply:
Ok. Removed the last commit to keep the two args instead. I'll come back to this later to reduce the number of args, likely with a new struct and a single argument.